src/Event/Http/HttpHandler.php

```diff
@@ -21,6 +21,6 @@ public function handle($event, Context $context): array
 
         $response = $this->handleRequest($httpEvent, $context);
 
-        return $response->toApiGatewayFormat($httpEvent->hasMultiHeader());
+        return $response->toApiGatewayFormat($httpEvent->hasMultiHeader(), (isset($event['version']) ? (float) $event['version'] : 1.0));
```

Comment:
This could probably be (I haven't tested):

```suggestion
        return $response->toApiGatewayFormat($httpEvent->hasMultiHeader(), $httpEvent->getPayloadVersion());
```

Reply:
Actually I'm thinking: if v2 doesn't support multiheaders at all, then `$httpEvent->hasMultiHeader()` should return `false` in that case, and we don't even need to know the payload version here?

That could simplify things even more.

Reply:
Good catch - now updated

Reply:
> Actually I'm thinking: if v2 doesn't support multiheaders at all, then `$httpEvent->hasMultiHeader()` should return `false` in that case, and we don't even need to know the payload version here?
> 
> That could simplify things even more.

For sure -- have fixed that up now too :)